ingest/ingest-publications.py

```diff
@@ -289,6 +291,74 @@ def create_publication_doc(pubgraph,publication):
     if volume:
         doc.update({"volume": volume})
 
+    number = list(pub.objects(predicate=BIBO.number))
+    volume = number[0].encode('utf-8') if number else None
+    if number:
+        doc.update({"articleNumber": number})
+
+    isbn10 = list(pub.objects(predicate=BIBO.isbn10))
+    isbn10 = isbn10[0].encode('utf-8') if isbn10 else None
+    if isbn10:
+        doc.update({"isbn10": isbn10})
+
+    isbn13 = list(pub.objects(predicate=BIBO.isbn13))
+    isbn13 = isbn13[0].encode('utf-8') if isbn13 else None
+    if isbn13:
+        doc.update({"isbn13": isbn13})
+
+    eissn = list(pub.objects(predicate=BIBO.eissn))
+    eissn = eissn[0].encode('utf-8') if eissn else None
+    if eissn:
+        doc.update({"eissn": eissn})
+
+    issn = list(pub.objects(predicate=BIBO.issn))
+    issn = issn[0].encode('utf-8') if issn else None
+    if issn:
+        doc.update({"issn": issn})
+
+    pageStart = list(pub.objects(predicate=BIBO.pageStart))
+    pageStart = pageStart[0].encode('utf-8') if pageStart else None
+    if pageStart:
+        doc.update({"pageStart": pageStart})
+
+    pageEnd = list(pub.objects(predicate=BIBO.pageEnd))
+    pageEnd = pageEnd[0].encode('utf-8') if pageEnd else None
+    if pageEnd:
+        doc.update({"pageEnd": pageEnd})
+
+#DRE    dataSource = list(pub.objects(predicate=PUBS.dataSource))
+#DRE    dataSource = dataSource[0].encode('utf-8') if dataSource else None
+#DRE    if dataSource:
+#DRE        doc.update({"dataSource": dataSource})
+
+    dateInCube = list(pub.objects(predicate=PUBS.dateInCube))
+    dateInCube = dateInCube[0].encode('utf-8') if dateInCube else None
+    if dateInCube:
+        doc.update({"dateInCube": dateInCube})
+
+    authorCount = list(pub.objects(predicate=PUBS.authorCount))
```

Comment:
Hey @delsborg , great work!, what kind of object is pub.objects(json?)? Also it looks like a function could be used for that along with some list comprehensions, I think that those are available on Python 2x

Reply:
Hi @ulgu3559 
pub in this context is a: https://rdflib.readthedocs.io/en/stable/_modules/rdflib/resource.html
A resource is like a specific node in a much larger graph.
The current paradigm for this ETL is to import the entire Publication/Author graph from CU Experts via SPARQL queries into a large union graph in python. It all fits by the way. The graph is the global g variable.
 The pub resource is a single publication with all of it's edges. Hence pub.objects returns I believe a generator of all objects for this resource that match a predicate, s in this case its a generator of 1 for authorCount.
 
 Once the entire index resides on AWS and this code runs in python 3.x I want to really simplify this, with list comprehensions and maybe even using pandas instead of or with rdflib.  Would love your help with that. Also to include testing.

Reply:
Really interesting, happy to help when you are ready.